docs/src/contributing/pull-request.md

```diff
@@ -0,0 +1,126 @@
+# The PR process for agave source code development
+
+## Preparation
```

Comment:
Should we include a pointer to https://github.com/solana-labs/solana?tab=readme-ov-file for environment setup?

Reply:
added reference to the relevant section in agave repo (it has slightly more details)